File: 1 lab/cpp/2.cpp
#include <iostream>

int main(int argc, char* argv[]) {

    int n;
    std::cin >> n;
    int m[n];
    for (int i = 0; i < n; i++) {
        std::cin >> m[i];
    }

    int x;
    std::cin >> x;

    int temp = INT_MAX;
    int res;
    
    for (int i : m) {
        if (abs(x - i) < temp) {
            temp = abs(x - i);
            res = i;
        }
    }

    std::cout << res << std::endl;

    return 0;
}

File: 1 lab/python/2.py
n = int(input())
m = list(map(int, input().split()))
m_dist = []
x = int(input())
temp = 10**8
res = 0
for i in m:
    if abs(x - i) < temp:
        temp = abs(x - i)
        res = i
print(res)